index.js

```diff
@@ -35,31 +46,101 @@ module.exports = {
+      this._cachedDebugTree = require('broccoli-debug').buildDebugCallback(`ember-cli-babel:${_parentName(this.parent)}`);
     }
 
     return this._cachedDebugTree.apply(null, arguments);
   },
 
+  /**
+   * Default babel options
+   * @param {*} config 
+   */
+  _getDefaultBabelOptions(config = {}) {
+     let emberCLIBabelConfig = config["ember-cli-babel"];
+     let providedAnnotation;
+     let throwUnlessParallelizable;
+     let sourceMaps = false;
+     let shouldCompileModules = _shouldCompileModules(config, this.project);
+
+     if (emberCLIBabelConfig) {
+       providedAnnotation = emberCLIBabelConfig.annotation;
+       throwUnlessParallelizable = emberCLIBabelConfig.throwUnlessParallelizable;
+     }
+ 
+     if (config.babel && "sourceMaps" in config.babel) {
+       sourceMaps = config.babel.sourceMaps;
+     }
+
+     let options = {
+       annotation: providedAnnotation || `Babel: ${_parentName(this.parent)}`,
+       sourceMaps,
+       throwUnlessParallelizable,
+       filterExtensions: _getExtensions(config, this.parent),
+       plugins: []
+     };
+
+     if (shouldCompileModules) {
+       options.moduleIds = true;
+       options.getModuleId = require("./lib/relative-module-paths").getRelativeModulePath;
+     }
+
+     options.highlightCode = _shouldHighlightCode(this.parent);
+     options.babelrc = false;
+     options.configFile = false;
+
+     return options;
+  },
+
   transpileTree(inputTree, _config) {
+
     let config = _config || this._getAddonOptions();
     let description = `000${++count}`.slice(-3);
     let postDebugTree = this._debugTree(inputTree, `${description}:input`);
-
-    let options = this.buildBabelOptions(config);
+    let options = this._getDefaultBabelOptions(config);
     let output;
-    if (this._shouldDoNothing(options)) {
+
+    const customAddonConfig = config['ember-cli-babel'];
+    const shouldUseBabelConfigFile = customAddonConfig && customAddonConfig.useBabelConfig;
+    
+    if (shouldUseBabelConfigFile) {
+      // Ref: https://github.com/babel/babel/blob/c6aea4e85d2b8f3e82575642d30b01c8cbe112a9/packages/babel-core/src/config/files/configuration.js#L24
+      // supported babel config root filenames.
+      const ROOT_CONFIG_FILENAMES = [
+        "babel.config.js",
+        "babel.config.cjs",
+        "babel.config.mjs",
+        "babel.config.json",
+      ].map(fileName => path.resolve(this.parent.root, fileName));
+      
+      const babelConfigFile = ROOT_CONFIG_FILENAMES.find((fileName) =>
+        fs.existsSync(fileName)
+      );
+      if (!babelConfigFile) {
+        throw new Error(
+          "Missing babel config file in the project root. Please double check if the babel config file exists or turn off the `useBabelConfig` option in your ember-cli-build.js file."
+        );
+      }
+      options = Object.assign({}, options, {
+        presets: [
+          require.resolve(babelConfigFile),
+        ],
+      });
```

Comment:
absolutely! for some unexplainable vague reason, the time I was developing this, I read in the docs to add in the file in presets 😑

Reply:
`loadPartialConfig` returns an object of [ConfigItems](https://babeljs.io/docs/en/babel-core/#configitem-type) which is not liked by broc transpiler since it expects a POJO. Since we are doing a hybrid approach where we are injecting our own options along with the config, one way we could solve this is by reading the config file inline and then pass it to the options.

Reply:
We chatted through this a bit in DM / calls, we are going to update to use `configFile` path from `.loadPartialConfig` which will properly be serializable and leave all of the identification and invocation of the config file up to babel itself.